Source/spells.cpp

```diff
@@ -294,48 +291,46 @@ void DoResurrect(int pnum, int rid)
 	}
 }
 
-void DoHealOther(int pnum, int rid)
+void DoHealOther(int pnum, uint16_t rid)
 {
-	int i, j, hp;
-
 	if (pnum == myplr) {
 		NewCursor(CURSOR_HAND);
 	}
 
+	if ((DWORD)pnum >= MAX_PLRS) {
+		app_fatal("CreatePlayer: illegal player %i", pnum);

```

Comment:
This should be `DoHealOther` instead of `CreatePlayer`. Having said that, I'm not sure that crashing is an appropriate response to receiving a bad packet. Seems like that could invite some foul play.